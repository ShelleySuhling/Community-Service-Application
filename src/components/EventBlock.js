import React, { Component } from 'react';
import { Card, Icon } from 'semantic-ui-react'

import moment from 'moment'

class ProfileForm extends Component {

    toDate = (date) => {
        //for whatever reason, firebase's .toDate() method of a timestamp wasn't working so I made my own
        return moment(date.seconds * 1000).format('MM/DD/YYYY')
    }

    toTime = (time) => {
        return moment(time.seconds * 1000).format('hh:mm A')
    }

    generateDescription = () => {
        let { event } = this.props
        return (
            <div className="card-container">
                <div className="card-info-container">
                    <div className="card-title">{event.title}</div>
                    <div className="card-description">
                        Location: {event.location}<br></br>
                        Date: {this.toDate(event.startTime)}<br></br>
                        Time: {this.toTime(event.startTime)} - {this.toTime(event.endTime)}
                    </div>
                </div>
                <div className="card-icons">
                    {event.isMandatory ? <Icon size='huge' color="blue" name='anchor' /> : null}
                </div>
            </div>
        )

    }

    render() {
        let { event } = this.props
        return <Card fluid color='blue' key={event.id} description={this.generateDescription()} />

    }

}

export default ProfileForm