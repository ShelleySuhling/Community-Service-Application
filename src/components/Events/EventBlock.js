import React, { Component } from 'react';
import { Link } from 'react-router-dom'
import { Card, Icon } from 'semantic-ui-react'
import moment from 'moment'

class EventBlock extends Component {

    generateDescription = () => {
        let { event, user } = this.props
        return (
            <div className="card-container">
                <div className="card-info-container">
                    <div className="card-title">
                        {event.title}
                        <div className="card-icons">
                            {/* {event.isMandatory ? <Icon color="blue" name='anchor' /> : null} */}
                        </div>
                    </div>
                    <div className="card-description">
                        <div><Icon color="blue" name='map marker alternate' /> {event.location}</div>
                        <div><Icon color="blue" name='calendar outline' /> {moment(event.startTime).format("MMM Do YY")}</div>
                        <div><Icon color="blue" name='clock outline' />{moment(event.startTime).format('LT')} - {moment(event.endTime).format('LT')}</div>
                    </div>
                </div>
                <div className="card-icon-container">
                    {/* <div className="card-icons">
                    </div> */}
                    {event.isMandatory
                        ? <div className="corner-icon">
                            <Icon size='huge' color="blue" name='caret up' />
                        </div>
                        : <div>{null}</div>}

                    {user.role === "master"
                        ? <Link to={"/edit_event/" + event.id}>
                            <Icon size='large' color="blue" name='pencil alternate' />
                        </Link>
                        : null
                    }

                </div>
            </div>
        )

    }

    render() {
        let { event } = this.props
        return <Card fluid color='blue' key={event.id} description={this.generateDescription()} />

    }

}

export default EventBlock