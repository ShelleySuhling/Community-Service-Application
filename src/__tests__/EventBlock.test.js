import * as React from 'react';
import { shallow } from 'enzyme';
import moment from 'moment'

import MockDate from 'mockdate'
import EventBlock from '../components/Events/EventBlock';

beforeAll(() => {
  MockDate.set('1/1/2018');
})

describe('EventBlock', () => {
  let mock_event = {
    id: 3,
    title: "sample title",
    location: "sample location",
    startDate: moment("2018-01-05 4:30", "YYYY-MM-DD HH:mm"),
    endDate: moment("2018-01-05 7:30", "YYYY-MM-DD HH:mm"),
    isMandatory: true,
  }
  it('matches snapshot', () => {
    const wrapper = shallow(<EventBlock event={mock_event} user={{ role: "master" }} />);
    expect(wrapper).toMatchSnapshot();
  });
});