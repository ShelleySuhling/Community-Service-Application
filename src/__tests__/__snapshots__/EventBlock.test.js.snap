// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventBlock matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EventBlock
    event={
      Object {
        "endDate": "2018-01-05T15:30:00.000Z",
        "id": 3,
        "isMandatory": true,
        "location": "sample location",
        "startDate": "2018-01-05T12:30:00.000Z",
        "title": "sample title",
      }
    }
    user={
      Object {
        "role": "master",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": "3",
    "nodeType": "class",
    "props": Object {
      "color": "blue",
      "description": <div
        className="card-wrapper"
      >
        <div
          className="card-info-container"
        >
          <div
            className="card-title"
          >
            sample title
            <div
              className="card-icons"
            />
          </div>
          <div
            className="card-description"
          >
            <div>
              <Icon
                as="i"
                color="blue"
                name="map marker alternate"
              />
               
              sample location
            </div>
            <div>
              <Icon
                as="i"
                color="blue"
                name="calendar outline"
              />
               
              Jan 1st 18
            </div>
            <div>
              <Icon
                as="i"
                color="blue"
                name="clock outline"
              />
              12:00 AM
               - 
              12:00 AM
            </div>
          </div>
        </div>
        <div
          className="card-icon-container"
        >
          <div
            className="corner-icon"
          >
            <Icon
              as="i"
              color="blue"
              name="caret up"
              size="huge"
            />
          </div>
          <Link
            replace={false}
            to="/edit_event/3"
          >
            <Icon
              as="i"
              color="blue"
              name="pencil alternate"
              size="large"
            />
          </Link>
        </div>
      </div>,
      "fluid": true,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": "3",
      "nodeType": "class",
      "props": Object {
        "color": "blue",
        "description": <div
          className="card-wrapper"
        >
          <div
            className="card-info-container"
          >
            <div
              className="card-title"
            >
              sample title
              <div
                className="card-icons"
              />
            </div>
            <div
              className="card-description"
            >
              <div>
                <Icon
                  as="i"
                  color="blue"
                  name="map marker alternate"
                />
                 
                sample location
              </div>
              <div>
                <Icon
                  as="i"
                  color="blue"
                  name="calendar outline"
                />
                 
                Jan 1st 18
              </div>
              <div>
                <Icon
                  as="i"
                  color="blue"
                  name="clock outline"
                />
                12:00 AM
                 - 
                12:00 AM
              </div>
            </div>
          </div>
          <div
            className="card-icon-container"
          >
            <div
              className="corner-icon"
            >
              <Icon
                as="i"
                color="blue"
                name="caret up"
                size="huge"
              />
            </div>
            <Link
              replace={false}
              to="/edit_event/3"
            >
              <Icon
                as="i"
                color="blue"
                name="pencil alternate"
                size="large"
              />
            </Link>
          </div>
        </div>,
        "fluid": true,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
